extensions/file-manager/src/manage-files.tsx

```diff
@@ -1,5 +1,36 @@
-import { getStartDirectory, Directory } from "./utils";
+import { useCallback, useRef, useState } from "react";
+import { Directory } from "./ui/Directory";
+import { getStartDirectory } from "./utils";
+import { AppContext, AppContextType } from "./utils/AppContext";
 
 export default function Command() {
-  return <Directory path={getStartDirectory()} />;
+  const [pathInContext, setPathInContext] = useState("");
+
+  const refShouldReRender = useRef(false);
+
+  const renderDirectory = useCallback((path: string) => {
+    setPathInContext(path);
+    refShouldReRender.current = true;
+  }, []);
+
+  const markFinishRenderingDirectory = useCallback(() => {
+    refShouldReRender.current = false;
+  }, []);
+
+  const shouldReRender = useCallback(() => {
+    return refShouldReRender.current;
+  }, []);
+
+  const value: AppContextType = {
+    path: pathInContext,
+    renderDirectory,
```

Comment:
I also added some comments to explain these props in this file `extensions/file-manager/src/utils/AppContext.tsx`